ethapi/api.go

```diff
@@ -1274,6 +1302,36 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 	return hexutil.Uint64(hi), nil
 }
 
+// getNumberAndTime returns the block number and time for the given block number or hash,
+// applying any overrides specified in blockOverrides.
+// if the number or hash is invalid or does not exist, an error is returned.
+func getNumberAndTime(ctx context.Context, b Backend, blockNrOrHash rpc.BlockNumberOrHash) (uint64, uint64, error) {
+
+	var header *evmcore.EvmHeader
+	var err error
+
+	// get header
+	rpcBlockNumber, ok := blockNrOrHash.Number()
+	if ok {
+		header, err = b.HeaderByNumber(ctx, rpcBlockNumber)
```

Comment:
Careful here: I think this function returns `nil` for the header and the error if no block is present with that number. This is probably unexpected, but it seems to be the convention in the RPC package.

You need to check the header for `nil` before accessing it, otherwise the code will panic.